registration/include/pcl/registration/impl/gicp.hpp

```diff
@@ -506,6 +711,7 @@ GeneralizedIterativeClosestPoint<PointSource, PointTarget>::computeTransformatio
       break;
     }
     nr_iterations_++;
+    is_translation_turn_ = !is_translation_turn_;
```

Comment:
If current iteration is optimizing translation part, next iteration will optimize rotation part. There may be a better way.